File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.cpp
#include <iostream>
#include<string>
#include<stack>
using namespace std;

int solution(string s)
{
    
    if(s.size()%2 == 1) return 0;
    
    stack<char> st;
    for(int i=0; i<s.size(); i++){
        if(!st.empty()&&st.top() == s[i]){
            st.pop();
        } else {
            st.push(s[i]);
        }
    }
    
    if(st.empty()) return 1;
    else return 0;

}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;

class Solution
{
    public int solution(String s)
    {
        int answer = 0;
        ArrayDeque<Character> st = new ArrayDeque<>();
        
        for(int i=0; i<s.length(); i++){
            if(!st.isEmpty() && st.peek() == s.charAt(i)){
                st.pop();
                continue;
            }
            st.push(s.charAt(i));
        }
        if(st.isEmpty()) answer = 1;
        
        return answer;
    }
}